plugins/techdocs/config.d.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export interface Config {
+  /** Configuration options for the techdocs plugin */
+  techdocs: {
+    /**
+     * attr: 'requestUrl' - accepts a string value
+     * e.g. requestUrl: http://localhost:7000/api/techdocs
+     * @visibility frontend
+     */
+    requestUrl: string;
+    /**
+     * attr: 'storageUrl' - accepts a string value
+     * e.g. storageUrl: http://localhost:7000/api/techdocs/static/docs
+     */
+    storageUrl: string;
```

Comment:
I thought this was optional? It should be possible to use only the `requestUrl` only if you don't use the backend. Same goes with `builder` below.

Reply:
`storageUrl` will be optional even for `techdocs-backend` [soon](https://github.com/backstage/backstage/issues/3715). Idea is the use `techdocs.publisher` for back-and-forth communication with storage.

`builder` is now required for the `techdocs-backend` as an explicit way of defining where the docs are being generated (local or external). More here https://backstage.io/docs/features/techdocs/configuration

I see your point that they can be moved over to `techdocs-backend` instead of `techdocs` (Although `builder` is used in the frontend to show better error message when docs are missing). Given that the current `techdocs` depends on `techdocs-backend` APIs for fetching documentation files, Metadata, to check if docs exist - it's now quite complex to use `techdocs` on its own without its backend. In future we will have limited access to some entities and their documentation for users, `techdocs-backend` becomes a necessity to handle these use cases for `techdocs`.

To be honest, I now see both plugin as a bundle for enabling TechDocs. :)

Reply:
Isn't one of the supported use cases is to not rely on the backend as per #2878. We currently do not use the `techdocs-backend` since it's all done with custom CI. Just want to make sure that our outlier use case is supported. Making these required suggests that you always need to use the `backend`. Granted there are already some assumptions about this that cause some odd UX (e.g. docs reader tries to build the site even though it can, but will render the site fine).

Reply:
> We currently do not use the techdocs-backend since it's all done with custom CI. Just want to make sure that our outlier use case is supported.

Generating docs on CI is quite the opposite of an outlier - we highlight it as the ["recommended" approach](https://backstage.io/docs/features/techdocs/architecture) due to its numerous benefits. However, `techdocs-backend` is not just about generating docs, and is also about the other use cases I mention above. You could ask `techdocs-backend` not to generate docs by setting `techdocs.builder` to `'external'`.

If you're using S3 to store the generated docs, you are already aware that `techdocs-backend` will soon support it soon. We are also working on making sure that the [CI process](https://github.com/backstage/backstage/issues/3097) for TechDocs is as smooth as possible.

Reply:
Ok sounds good. I've been neglecting the docs side our our backstage instance, and need to keep up with everything going on 😄 Just want to make sure nothing is broken before we get the chance to migrate over to the newer patterns.

Reply:
@andrewthauer More than happy to help you use the newer patterns. And thank you for engaging. :)
One of our objectives for next quarter is to write lots of documentation and HOWTO guides for TechDocs. So, expect a lot of support. 😁